src/coreclr/tools/Common/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -28,6 +28,7 @@ public sealed partial class EcmaType : MetadataType, EcmaModule.IEntityHandleObj
         private TypeDesc[] _genericParameters;
         private MetadataType _baseType;
         private int _hashcode;
+        private MethodImplRecord[] _allMethodImplsForType;
```

Comment:
We should inline this logic to where it's needed (or into an extension method). It doesn't look like we need to cache this and metadata reader + handle is available as a public API on `EcmaType`.